app/models/katello/host/content_facet.rb

```diff
@@ -99,6 +100,15 @@ def single_lifecycle_environment
         content_view_environments&.first&.lifecycle_environment
       end
 
+      def content_view_environments=(new_cves)
+        super(new_cves)
+        return unless new_cves.length > 1
+        Rails.logger.debug("ContentFacet: Setting CVEs for host #{host.name}: #{new_cves.map(&:candlepin_name).join(', ')}")
+        Katello::ContentViewEnvironmentContentFacet.reprioritize_for_content_facet(self, new_cves)
+        self.content_view_environments.reload
```

Comment:
Do we need to reload this?

Reply:
It wasn't working for me without reloading.